src/pcre2_jit_compile.c

```diff
@@ -3184,7 +3185,7 @@ while (cc < ccend)
       SLJIT_ASSERT(private_srcw[i] != 0);
 
       if (!from_sp)
-        delayed_mem_copy_move(&status, base_reg, stackptr, SLJIT_SP, private_srcw[i]);
+        delayed_mem_copy_move(&status, base_reg, stackptr, SLJIT_SP, (int)private_srcw[i]); // XXX @zherczeg: cast OK?
```

Comment:
@zherczeg I have added three casts here. I cannot verify whether these are OK. They are all truncation of an `sljit_sw` to an int.

It appears that all of the data placed inside kept_shared_srcw is within the range of an int; but private_srcw and shared_srcw are harder to check.

I'm hoping you "just know" already that these assignments are safe.

Reply:
`sljit_s32` could be better than int. The casts are ok, this should copy between (machine) stack and (backtracking) stack :)

Reply:
Thank you very much for helping me out!

I have changed some uses of `int` to `sljit_s32`.

I think the changes to `pcre2_jit_compile` in this PR are now simply formatting - all the variables have the same type as before (since `int` and `sljit_s32` are actually the same), and we've just added explicit casts for things that were previously implicit casts. So the runtime code should be identical.

Reply:
> I have changed some uses of `int` to `sljit_s32`.

historically (in the long obsolete PCRE1 codebase) all the interfaces used `int` and some might had not been updated since, so is usually safe to make this "conversion", but should also probably look for variables that were really meant to become PCRE2_SIZE, or in the case of JIT one of sljit_sw/sljit_uw.

Reply:
Yes Carlo, I _did_ look for cases where it should be PCRE2_SIZE or sljit_sw. That's what the discussion was about. I only changed values to sljit_s32 because Zoltan suggested it was better.

Reply:
PCRE1 used int because, being started in 1997, it was written to the C90 standard, in which int32_t etc did not exist. (Just for the record.)